addons/binding/org.openhab.binding.fileregexparser/src/main/java/org/openhab/binding/fileregexparser/handler/FileRegexParserHandler.java

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fileregexparser.handler;
+
+import static org.openhab.binding.fileregexparser.FileRegexParserBindingConstants.CHANNEL_GROUPCOUNT;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.fileregexparser.internal.FileRegexParserWorker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FileRegexParserHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author meju25 - Initial contribution
+ */
+public class FileRegexParserHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(FileRegexParserHandler.class);
+    private String fileName;
+    private String regEx;
+    private static Pattern pattern;
+    private static Matcher matcher;
```

Comment:
I do not remember, but I think I took this from an example in the Internet.
To be honest, I am not an expert in Java. So if you say it is not necessary and dangerous I will remove the static keyword.